SecretSanta/test/SecretSanta.Api.Tests/Controllers/UsersControllerTests.cs

```diff
@@ -0,0 +1,236 @@
+            Assert.IsTrue(result.Result is NotFoundResult);
+            Assert.IsTrue(result2.Result is NotFoundResult);
+        }
+
+        [TestMethod]
+        [DataRow(3)]
+        public void Delete_WithGoodID_ReturnsProperCount(int id)
+        {
+            //Arrange
+            UserRepository repository = new();
+            UsersController controller = new(repository);
+            IEnumerable<User> ExpectedCount = controller.Get();
+            int expected = ExpectedCount.Count<User>()-1;
+ 
+    
+
+            //Act
+            controller.Delete(id);
+            IEnumerable<User> Final = controller.Get();
+            int expected2 = Final.Count<User>();
+            Boolean res = (expected == expected2);
+
+            //Assert
+            Assert.IsTrue(res);
+        }
+        [TestMethod]
+        public void Post_WithGoodData_ReturnsListWithNewUser()
+        {
+            User newUser = new User();
+            newUser.FirstName = "New";
+            newUser.LastName = "Mann";
+            //Arrange
+            UserRepository repository = new();
+            UsersController controller = new(repository);
+            IEnumerable<User> List = controller.Get();
+ 
+    
+
+            //Act
+            controller.Post(newUser);
+            Boolean res = (List.Contains<User>(newUser));
+
+            //Assert
+            Assert.IsTrue(res);
+        }
+
+        [TestMethod]
+        public void Post_WithBadData_ReturnsListWithoutNewUser()
+        {
+            User newUser = null;
+            //Arrange
+            UserRepository repository = new();
+            UsersController controller = new(repository);
+            IEnumerable<User> List = controller.Get();
+ 
+    
+
+            //Act
+            controller.Post(newUser);
+            Boolean res = (!(List.Contains<User>(newUser)));
+
+            //Assert
+            Assert.IsTrue(res);
+        }
+
+        [TestMethod]
+        [DataRow(3)]
+        public void Put_WithGoodData_ReturnsListWithUser(int id)
+        {
+            UpdateUser update = new UpdateUser();
+            update.FirstName = "New";
+            update.LastName = "Mann";
+            //Arrange
+            UserRepository repository = new();
+            UsersController controller = new(repository);
+            
+ 
+    
+
+            //Act
+            controller.Put(id, update);
+            IEnumerable<User> List = controller.Get();
+            Boolean res = (List.ElementAt<User>(id-1).FirstName==update.FirstName && List.ElementAt<User>(id-1).LastName==update.LastName);
+
+            //Assert
+            Assert.IsTrue(res);
+        }
+
+        [TestMethod]
+        [DataRow(3)]
+        public void Put_WithBadData_ReturnsListWithoutUser(int id)
+        {
+            UpdateUser update = null;
+            //Arrange
+            UserRepository repository = new();
+            UsersController controller = new(repository);
+            Boolean res = false;
```

Comment:
Nit: Favor using C# keywords over type names when using privative types (bool, int, string, etc). 
```suggestion
            bool res = false;
```